test/functorch/test_aotdispatch.py

```diff
@@ -6022,6 +6022,101 @@ def fn(x):
         out = torch.compile(fn, backend="aot_eager", fullgraph=True)(inp)
         self.assertEqual(ref_out, out)
 
+    def test_channels_last_grads_no_force_contiguous_dense(self):
+        class M(torch.nn.Module):
+            def __init__(self) -> None:
+                super().__init__()
+                self.conv = torch.nn.Conv2d(3, 3, 3)
+
+            def forward(self, x, y):
+                z = y + 3
+                y.mul_(2)
+                r = self.conv(x)
+                return r, r.transpose(0, 1), z.view(-1), z.transpose(0, 1)
+
+        m = M()
+        m.to(memory_format=torch.channels_last)
+        m.train()
+
+        def dense_inps():
+            ret = (
+                torch.randn(2, 3, 5, 5, requires_grad=True).to(
+                    memory_format=torch.channels_last
+                ),
+                torch.randn(3, 2, 1, 1, requires_grad=True).to(
+                    memory_format=torch.channels_last
+                ),
+            )
+            [inp.retain_grad() for inp in ret]
```

Comment:
why are all of the retain_grad calls necessary throughout these tests?

Reply:
Yeah, I am not doing grad assert checks here, we can remove it.